src/backend/distributed/commands/function.c

```diff
@@ -224,7 +224,10 @@ create_distributed_function(PG_FUNCTION_ARGS)
 	 * pg_dist_object, and not propagate the CREATE FUNCTION. Function
 	 * will be created by the virtue of the extension creation.
 	 */
-	if (IsObjectAddressOwnedByExtension(&functionAddress, &extensionAddress))
+	ObjectAddress *copyFunctionAddress = palloc0(sizeof(ObjectAddress));
```

Comment:
can we avoid `copyFunctionAddress` ? It is simpler to adjust `EnsureExtensionFunctionCanBeDistributed` to get a pointer